src/osyris/spatial/subdomain.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/osyris-project/osyris)
+
+import numpy as np
+
+
+def extract_sphere(dataset, radius, origin):
+    """
+    Extract a spherical subdomain around an origin point.
+    """
+    subdomain = dataset.__class__(nout=dataset.meta["nout"], path=dataset.meta["path"])
+
+    for name, group in dataset.items():
+        pos = group.get("position", group.parent["amr"]["position"])
+        r = (pos - origin).norm
+        c = (r < radius).values
+        if np.any(c):
+            subdomain[name] = dataset[name][c]
+
+    return subdomain
+
+
+def extract_box(dataset, dx, dy, dz, origin):
+    """
+    Extract a cubic domain of size dx, dy & dz around an origin point
+    """
+    subdomain = dataset.__class__(nout=dataset.meta["nout"], path=dataset.meta["path"])
+
+    for name, group in dataset.items():
+        pos = group.get("position", group.parent["amr"]["position"])
+        centered_pos = pos - origin
+        cx = (centered_pos.x <= dx * .5) & (centered_pos.x >= -dx * .5)
+        cy = (centered_pos.y <= dy * .5) & (centered_pos.y >= -dy * .5)
+        cz = (centered_pos.z <= dz * .5) & (centered_pos.z >= -dz * .5)
+        c = (cx & cy & cz).values
+        if np.any(c):
+            subdomain[name] = dataset[name][c]
+
+    return subdomain
```

Comment:
Can we also add a `extract_plane` function?
Then the `map` code will in the end be able to just extract a subdomain, rotate it, and then we are done :-)

Reply:
I'll think about how to do this!

Reply:
Could I implement this in a follow up PR?